titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/JobManagerUtil.java

```diff
@@ -148,6 +149,15 @@ public static Task attachPlacementData(Task task, Optional<TitusExecutorDetails>
         }).orElse(task);
     }
 
+    public static Task attachKubeletData(Task task, PodWrapper podWrapper) {
+        String ipv4 = podWrapper.getV1Pod().getStatus().getPodIP();
+        Map<String, String> newContext = new HashMap<>(task.getTaskContext());
+        BiConsumer<String, String> contextSetter = (key, value) -> StringExt.applyIfNonEmpty(value, v -> newContext.put(key, v));
```

Comment:
You can use equivalent method from Evaluators class in titus-common.

Reply:
I see `applyNotNull` in Evaulators, but that won't help me against Empty strings though.